src/sonic-bgpcfgd/bgpmon/bgpmon.py

```diff
@@ -135,19 +134,19 @@ def update_neigh_states(self):
             if len(data) > PIPE_BATCH_MAX_COUNT:
                 self.flush_pipe(data)
         # Check for stale state entries to be cleaned up
-        while len(self.peer_l) > 0:
+        for peer in self.peer_l:
             # remove this from the stateDB and the current neighbor state entry
-            peer = self.peer_l.pop(0)
             del_key = "NEIGH_STATE_TABLE|%s" % peer
             data[del_key] = None
-            del self.peer_state[peer]
+            if peer in self.peer_state:
+                del self.peer_state[peer]
             if len(data) > PIPE_BATCH_MAX_COUNT:
                 self.flush_pipe(data)
         # If anything in the pipeline not yet flushed, flush them now
         if len(data) > 0:
             self.flush_pipe(data)
-        # Save the new List
-        self.peer_l = self.new_peer_l[:]
+        # Save the new dict
```

Comment:
Overall the change looks great.  Just the comment here needs to be changed from "dict" to "set" to avoid confusion.  Nice work overall!  Thanks for the fix!

Reply:
Thanks. Updated.
So I'll close #6547. I think it's not an issue. Correct?

Reply:
@bingwang-ms Yes. #6547 is not an issue. Your PR fix will not allow duplicated entry in STATE DB to occur.
Thanks!